zip-0208.rst

```diff
@@ -0,0 +1,222 @@
+::
+
+  ZIP: 208
+  Title: Shorter Block Target Spacing
+  Author: Simon Liu <simon@z.cash>
+          Daira Hopwood <daira@z.cash>
+  Category: Consensus
+  Created: 2018-01-10
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", and "MAY" in this document are to
+be interpreted as described in RFC 2119. [#RFC2119]_
+
+The terms "block chain", "consensus rule change", "branch", and "network
+upgrade" are to be interpreted as defined in [#zip-0200]_.
+
+The term "block target spacing" means the time interval between blocks targeted
+by the difficulty adjustment algorithm in a given branch. It is normally
+measured in seconds. (This is also sometimes called the "target block time",
+but "block target spacing" is the term used in [#latest-protocol]_.)
+
+
+Abstract
+========
+
+This proposal specifies a change in the block target spacing, to take effect in
+the Blossom network upgrade [#zip-0206]_.
+
+The emission schedule of mined ZEC will be approximately the same in terms of
+time, but this requires the emission per block to be adjusted to take account
+of the changed block target spacing.
+
+
+Motivation
+==========
+
+The motivations for decreasing the block target spacing are:
+
+- Reduced latency for considering transactions to be sufficiently confirmed;
+- Greater throughput of transactions in unit time.
+
+The latter goal could alternatively be achieved by increasing the block size
+limit, but that would not also achieve the former goal.
+
+Note that, for a given security requirement (in terms of the expected cost
+distribution of a rollback attack), the number of confirmations needed
+increases more slowly than the decrease in block time, and so, up to a point,
+decreasing the block target spacing can provide a better trade-off between
+latency and security. This argument assumes that the verification and
+propagation time for a block remain small compared to the block target spacing.
+
+
+Specification
+=============
+
+The changes described in this section are to be made in [#latest-protocol]_,
+relative to the pre-Blossom specification in [#preblossom-protocol].
+
+Throughout the specification, rename HalvingInterval to PreBlossomHalvingInterval,
+and rename PoWTargetSpacing to PreBlossomTargetSpacing. These constants retain
+their values from [#preblossom-protocol]_ of 840000 (blocks) and 150 (seconds)
+respectively.
+
+In section 2 (Notation), add BlossomActivationHeight and PostBlossomPoWTargetSpacing
+to the list of integer constants.
+
+In section 5.3 (Constants), define PostBlossomPoWTargetSpacing := 75 seconds.
+
+TODO: check that PostBlossomPoWTargetSpacing is reasonable.
```

Comment:
Yes it was in scope of the audits. Coinspect said:

> ZIP-208 - Consider Worst Case Block Verification Time
> The time between blocks must not be too close to the worst case of block verification time. The proposed block target spacing is 75 seconds and we believe it could be too close to the worst case verification time.

I think we need to make an estimate of worst-case verification time with parallelization (and with batching *if* we intend to rely on implementing that for all transactions in a block).

NCC did not say anything about the advisability of choosing 75s, although they went into much more detail than Coinspect about difficulty adjustment and clock skew issues.

The effect on the recommended number of confirmations is covered by #215. The verification time issue is discussed on https://github.com/zcash/zcash/issues/3690 .